### PR TITLE
Added setup_complete method to monitor maker setup

### LINKED ISSUE #104: [enhance] State variable in Maker indiicating full setup.
As mentioned here https://github.com/utxo-teleport/teleport-transactions/pull/76#issuecomment-1976858128

We need a state variable in the Maker struct to identify when a maker is fully set up with all the scanner threads and fidelity bond confirmation. A maker is fully set up when the fidelity bond transaction is confirmed. In reality too, this will be the task that takes the longest.

So the job is simple. Once the fidelity bond is confirmed, mark the state as "fully-setup".

Use this marker in integration tests to wait for full setup of the makers and then only make the balance query.

### PR DESCRIPTION
**Pull request by Shourya742**
Closes #104

### CODE REVIEW

File: tests/abort1.rs [hunk 1]
```diff
@@ -127,7 +127,16 @@ async fn test_stop_taker_after_setup() {
         .collect::<Vec<_>>();
 
     // Start swap
-    thread::sleep(Duration::from_secs(360)); // Take a delay because Makers take time to fully setup.
+
+    // Makers take time to fully setup.
+    makers.iter().for_each(|maker| {
+        while !*maker.is_setup_complete.read().unwrap() {
+            // Introduce a delay of 10 units to prevent write lock starvation.
+            thread::sleep(Duration::from_secs(10));
```
Comment (rajarshimaitra) [inline tests/abort1.rs:135]: One info log here like "waiting for maker setup completion" would be helpful.

**Review by rajarshimaitra — CHANGES_REQUESTED** [review]
ACK

File: tests/abort1.rs (hunk 1 above)
Comment (Shourya742) [reply tests/abort1.rs:135]: Cool will resolve this in next commit

**Review by rajarshimaitra — APPROVED** [review]
ACK 99d4c97ab076be069048082050aadb416fd6b72f

**Pull request merged**